lib/t/matches.ex

```diff
@@ -894,4 +895,65 @@ defmodule T.Matches do
     )
     |> where([m, e, c], is_nil(c.timestamp))
   end
+
+  def check_matches_calls() do
+    matches_without_successful_calls =
+      Match
+      |> join(:inner, [m], c in Call,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
+      )
+      |> where([m, c], c.accepted_at >= fragment("now() - interval '80 hours'"))
+      |> select([m], m.id)
+
+    matches_for_push =
+      Timeslot
+      |> where([t], t.selected_slot >= fragment("now() - interval '80 hours'"))
+      |> where([t], t.match_id not in subquery(matches_without_successful_calls))
+      |> select([t], t.match_id)
+
+    good_users =
+      Call
+      |> join(:inner, [c], m in Match,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
```

Comment:
all of these is super messy
if needed, just add match_id column to calls table (just keep in mind that there is an edge case of user calling back from a "missed call" when they are already not matched with the mater -> match_id will be nil here)